shibata added English to some eadme.md and translated vantiq-install-closed-network-aws.md and fluentd.md (#24)
fujitake/vantiq-related: vantiq-devices-integration/docs/eng/fluentd.md

# Data integration to Vantiq using Fluentd

> It is possible to integrate the metrics data acquired by the device into Vantiq using fluentd if the data is output in flat file format to log files, etc. In this article, it explains how to configure and integrate data using nginx access logs as an example.  

Reviewer: ```suggestion
Fluentd is useful to send the telemetry data to Vantiq if the data are generated in a flat file format like log files, etc. This article explains how to configure and integrate data, using nginx access logs as an example.  
```

> ## How to configure Fluentd

Reviewer: ```suggestion
In this article, assume the folllowing:
```

> - Assuming **Windows 10, Linux (Ubuntu 20.04)** as OS

Reviewer: ```suggestion
- Use **Windows 10, Linux (Ubuntu 20.04)** as OS
```

- Use Vantiq as a method to verify operation (receive Topic)

> - An example of transforming a flat file (nginx log) using regular expressions  

Reviewer: ```suggestion
- Use regular expressions to transform a flat file (nginx log)
```

<h3 id="dlubuntu">2.1 Download from the following site</h3>

- There are amd64 version and arm64 version, so choose the appropriate one for the environment.
- [Download Fluentd](https://www.fluentd.org/download)
- Ubuntu **Focal**.  
<img src="../../imgs/fluentd/td-agent-ubuntu.jpg" width="70%">

- Move to the folder where it was downloaded and install it.  
```
cd ~/Downloads
sudo dpkg -i ./td-agent_4.2.0-1_amd64.deb
```
<h3 id="plugin-ubnt">2.2 Install Plugin (http_ext)</h3>

```
cd /opt/td-agent
sudo td-agent-gem install fluent-plugin-out-http-ext
```

<h3 id="conf-ubnt">2.3 Create a configuration file</h3>

- Create the **/opt/td-agent/td-agent.conf** file with an appropriate editor and copy and paste the following contents into it.   
- Note that this configuration assumes the following contents.  

|Item|Details|
---|---
|①: Vantiq Topic name|The Resource URL of the Topic that receives events on Vantiq side.<br /> The following example corresponds to `jp.co.vantiq.fluentd`.|
|②: Access Token|Access Token issued in the Namespace of Vantiq target.|
|③: path|Path and file name of the log to monitor (in this case, nginx).|

> |④: pos_file|Path and file name for counting the log files to monitor.|

Reviewer: ```suggestion
|④: pos_file|Path and file name for keeping track of the position in the target log files.|
```

```
####
## Output descriptions:
##
<match **>
  @type http_ext
  endpoint_url      https://internal.vantiqjp.com/api/v1/resources/topics//jp.co.vantiq/fluentd # ①
  http_method       post
  serializer        json
  use_ssl           true
  verify_ssl        false
  <headers>
    Authorization Bearer dWXRMtWjRXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX6EtrM= # ②
  </headers>
</match>

####
## Source descriptions:
##
<source>
  @type tail
  path     /var/log/nginx/access.log # ③
  pos_file /var/log/nginx/access.log.pos  # ④
  tag local.syslog
  <parse>
    @type none
  </parse>
</source>
```

<h3 id="invoke-ubnt">2.4 Start Fluentd</h3>

```
cd /opt/td-agent
sudo bin/fluentd -c share/td-agent.conf
```

<h3 id="confirm-ubnt">2.5 Verify the operations</h3>

- Confirm that it is possible to access the nginx site and receive its access log.  

|Topic|Received data|
---|---
|<img src="../../imgs/fluentd/result.jpg">|<img src="../../imgs/fluentd/tskevent-ubnt.jpg">|

<h2 id="regexp">3. Aggregating logs using regular expressions（Vantiq IDE）</h2>

> In this section, describe how to transform the event data received in Vantiq's Topic.  

Reviewer: ```suggestion
This section describes how to transform the event data received in Vantiq's Topic.  
```
